lib/sendgrid_api/delivery_methods/sendgrid.rb

```diff
@@ -103,6 +104,22 @@ def parse_xsmtpapi_headers(mail)
     #
     def parse_email_address(email)
       Array(email).collect { |email| Mail::Address.new(email) }
+    rescue Mail::Field::ParseError => e
+      punycode_email(email)
     end
+
+    # convert email domain names containing unicode characters to punycode
+    # @param [String] email
+    # @return [Array] Mail::Address
+    #
+    def punycode_email(email)
+      Array(email).collect do |email|
+        email_part, display_name = email.split(/\<(.*?)\>/).reverse
```

Comment:
Interesting use of split! Suggest pattern improvements: `/\s*<(.+?)>\z/`

* anchor at end of string
* expect optional whitespace before `<` in pattern match
* don't need backslashes to escape `<>`
* expect 1 or more characters in `<>`